File: BACKEND/SCRIPT/LIB/loader.php
<?php

# FILE: 	load.php
# OWNER: 	DESAPHY JEREMY
# DATE:		05.30.2019
# PURPOSE:	master file to load all other processing files
 use PHPMailer\PHPMailer\PHPMailer;
 if (is_file('/composer/vendor/autoload.php'))require_once '/composer/vendor/autoload.php';
    

function microtime_float()
{
    list($usec, $sec) = explode(" ", microtime());
    return ((float)$usec + (float)$sec);
}
$START_SCRIPT_TIME=microtime_float();
$START_SCRIPT_TIMESTAMP=date('Y-m-d H:i:s');


$PROCESS_CONTROL=array('STEP'=>0,
		       'JOB_NAME'=>@$JOB_NAME,
			'DIR'=>'',
		       'LOG'=>array(),
		       'STATUS'=>'INIT',
		       'START_TIME'=>microtime_float(),
		       'END_TIME'=>'',
		       'STEP_TIME'=>microtime_float(),
		       'FILE_LOG'=>''
		    );



$time=microtime_float();
$TG_DIR= getenv('TG_DIR');

if ($TG_DIR===false)  sendKillMail('000001','NO TG_DIR found');
if (!is_dir($TG_DIR)) sendKillMail('000002','TG_DIR value is not a directory '.$TG_DIR);
if (!is_dir($TG_DIR.'/PROCESS') && !mkdir($TG_DIR.'/PROCESS'))sendKillMail('000002','TG_DIR/PROCESS can\'t be created');

$FILE_TO_LOAD=array(
	'/LIB/global.php',
	'/LIB/fct_utils.php',
	'/LIB/loader_process.php',
	'/LIB/loader_timestamp.php',
	'/LIB/loader_qengine.php',
	'/LIB/webjob_utils.php',
	
);


foreach ($FILE_TO_LOAD as $FILE)
{
	$PATH=$TG_DIR.'/BACKEND/SCRIPT/'.$FILE;

	$time=microtime_float();
	if ((include $PATH)==TRUE)continue;

	sendKillMail('000003','Unable to load file: '.$PATH);
}


if (defined("MONITOR_JOB"))
{
	foreach ($GLB_TREE as $ID=>&$INFO)
	{
		if (substr($INFO['NAME'],0,4)!='rmj_')continue;
		$PATH=$TG_DIR.'/BACKEND/';
		if ($INFO['IS_PRIVATE'])$PATH.='PRIVATE_SCRIPT/';
		else $PATH.='SCRIPT/';
		$PATH.=$INFO['DIR'].'/'.$INFO['NAME'].'.php';
		if (!checkFileExist($PATH))sendKillMail('000004','Unable to locate file: '.$PATH);
		if ((include $PATH)==TRUE)continue;

		sendKillMail('000005','Unable to load file: '.$PATH);
		
	}
}

function sendMail($ERROR_ID,$INFO)
{
	echo "SEND MAIL\n";
	echo $INFO."\n";
	global $GLB_VAR;
	global $PROCESS_CONTROL;
	
	if (isset($GLB_VAR['EMAIL']) && isset($GLB_VAR['EMAIL_FROM']))
	{

		$tab=explode("|",$GLB_VAR['EMAIL']);
		try{
			// create a new object
			$mail = new PHPMailer();
			// configure an SMTP
			$mail->isSMTP();
			// $mail->Host = 'live.smtp.mailtrap.io';
			// $mail->SMTPAuth = true;
			// $mail->Username = '1a2b3c4d5e6f7g';
			// $mail->Password = '1a2b3c4d5e6f7g';
			// $mail->SMTPSecure = PHPMailer::ENCRYPTION_STARTTLS;
			// $mail->Port = 2525;
			
			$mail->setFrom($GLB_VAR['EMAIL_FROM'], 'BIORELS - '.$ERROR_ID);
			foreach ($tab as $EM)
			$mail->addAddress($EM, 'Me');
			$mail->Subject = 'Issue';
			// Set HTML 
			$mail->isHTML(TRUE);
			$mail->Body = $INFO.' '.print_r($PROCESS_CONTROL,true);;
			$mail->AltBody = $INFO;
		   
			
			// send the message
			if(!$mail->send()){
				echo 'Message could not be sent.';
				echo 'Mailer Error: ' . $mail->ErrorInfo;
			} else {
				echo 'Message has been sent';
			}
			$mail=null;
		}catch(Exception $e)
		{}
	}
	
}
function sendKillMail($ERROR_ID,$INFO)
{
	echo "SEND KILL MAIL\n";
	sendMail($ERROR_ID,$INFO);
	exit(1);
}

date_default_timezone_set($GLB_VAR['TIMEZONE']);


$DB_CONN=null;
$DB_INFO=array();
$GLB_VAR['DB_SCHEMA']=getenv('DB_SCHEMA');
$GLB_VAR['SCHEMA_PRIVATE']=getenv('SCHEMA_PRIVATE');

connectDB();


loadTimestamps();

?>

File: BACKEND/SCRIPT/LIB_PYTHON/loader.py
import os
import time
from datetime import datetime
import smtplib
import psycopg2
import pickle
import sys
from webjob_utils import preload_web_jobs
import fct_utils
from fct_utils import *

def microtime_float():
    return time.time()
START_SCRIPT_TIME = microtime_float()
START_SCRIPT_TIMESTAMP = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
# PROCESS_CONTROL = {
#     'STEP': 0,
#     'JOB_NAME': JOB_NAME,  # Assuming JOB_NAME is defined somewhere
#     'DIR': '',
#     'LOG': [],
#     'STATUS': 'INIT',
#     'START_TIME': microtime_float(),
#     'END_TIME': '',
#     'STEP_TIME': microtime_float(),
#     'FILE_LOG': ''
# }
time_t = microtime_float()
if TG_DIR is False:
    send_kill_mail('000001', 'NO TG_DIR found')
if not os.path.isdir(TG_DIR):
    send_kill_mail('000002', 'TG_DIR value is not a directory ' + TG_DIR)
if not os.path.isdir(TG_DIR + '/PROCESS') and not os.makedirs(TG_DIR + '/PROCESS'):
    send_kill_mail('000002', 'TG_DIR/PROCESS can\'t be created')
loadProcess()
if ('MONITOR_JOB' in locals()):
	checkTree()
	genHierarchy()
time_t = microtime_float()
os.environ['TZ'] = GLB_VAR['TIMEZONE']
time.tzset()
GLB_VAR['DB_SCHEMA'] = os.getenv('DB_SCHEMA')
GLB_VAR['SCHEMA_PRIVATE'] = os.getenv('SCHEMA_PRIVATE')
connect_db()

load_timestamps()